runelite-client/src/main/java/net/runelite/client/util/LinkBrowser.java

```diff
@@ -0,0 +1,109 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.util;
+
+import java.awt.Desktop;
+import java.awt.Toolkit;
+import java.awt.datatransfer.StringSelection;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.client.ui.ClientUI;
+
+/**
+ * Utility class used for browser navigation
+ */
+@Singleton
+@Slf4j
+public class LinkBrowser
+{
+	private final Provider<ClientUI> clientUIProvider;
+
+	@Inject
+	private LinkBrowser(final Provider<ClientUI> clientUIProvider)
+	{
+		this.clientUIProvider = clientUIProvider;
+	}
+
+	/**
+	 * Tries to navigate to specified URL in browser. In case operation fails, displays message box with message
+	 * and copies link to clipboard to navigate to.
+	 * @param url url to open
+	 * @return true if operation was successful
+	 */
+	public boolean browse(final String url)
+	{
+		if (!Desktop.isDesktopSupported())
+		{
+			showMessageBox("Desktop is not supported. Link was copied to your clipboard.", url);
+			return false;
+		}
+
+		final Desktop desktop = Desktop.getDesktop();
+
+		if (!desktop.isSupported(Desktop.Action.BROWSE))
+		{
+			showMessageBox("Desktop browser is not supported. Link was copied to your clipboard.", url);
+			return false;
+		}
+
+		try
+		{
+			desktop.browse(new URI(url));
+			log.debug("Opened browser to {}", url);
+			return true;
+		}
+		catch (IOException | URISyntaxException ex)
+		{
+			log.warn("Unable to open specified url", ex);
+			showMessageBox("Unable to open specified url. Link was copied to your clipboard.", url);
```

Comment:
The user does not know what URL is "specified". Maybe "Unable to open a URL. Link was copied to your clipboard." is better.

Reply:
Ah, right, true.